app/api/webhooks/route.js

```diff
@@ -0,0 +1,31 @@
+// The Replicate webhook is a POST request where the request body is a prediction object.
+// Identical webhooks can be sent multiple times, so this handler must be idempotent.
+
+import { NextResponse } from 'next/server';
+import { validateWebhook } from 'replicate';
+
+export async function POST(request) {
+  console.log("Received webhook...");
+  
+  const secret = process.env.REPLICATE_WEBHOOK_SIGNING_SECRET;
+
+  if (!secret) {
+    console.log("Skipping webhook validation. To validate webhooks, set REPLICATE_WEBHOOK_SIGNING_SECRET")
+    const body = await request.json();
+    console.log(body);
+    return NextResponse.json({ detail: "Webhook received (but not validated)" }, { status: 200 });
+  }
+  
+  const webhookIsValid = await validateWebhook(request.clone(), secret);
```

Comment:
I'm doing this `request.clone()` business here because you can only read the request body once in Next.js API routes. Without this change, you'll get `TypeError: Body is unusable` when calling `request.json()` after validation.

@mattt what do you think of this?